ds3-contract-comparator-print/src/main/java/com/spectralogic/ds3contractcomparator/print/htmlprinter/generators/row/ModifiedHtmlRowGenerator.java

```diff
@@ -0,0 +1,244 @@
+        return getPropertyName(oldObject, newObject);
+    }
+
+    /**
+     * Retrieves the name of the unique property which is used to determine
+     * if two instances of a class are representing the same contract
+     * element. Either {@code oldObject} or {@code newObject} can be null,
+     * but not both.
+     * @param oldObject The old contract's version of the object. May be
+     *                  null if object does not exist in the old contract.
+     * @param newObject The new contract's version of the object. May be
+     *                  null if object does not exist in the new contract.
+     */
+    static <T> String getPropertyName(final T oldObject, final T newObject) {
+        return oldObject == null ? getUniqueProperty(newObject) : getUniqueProperty(oldObject);
+    }
+
+    /**
+     * Creates a union of all values for the specified property in both lists of objects
+     */
+    static <T> ImmutableSet<String> toPropertyUnion(
+            final ImmutableList<T> oldList,
+            final ImmutableList<T> newList,
+            final String property) {
+        final ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        builder.addAll(getPropertyValues(oldList, property));
+        builder.addAll(getPropertyValues(newList, property));
+        return builder.build();
+    }
+
+    /**
+     * Gets the set of property values from a list of objects
+     */
+    static <T> ImmutableSet<String> getPropertyValues(final ImmutableList<T> list, final String property) {
+        if (isEmpty(list)) {
+            return ImmutableSet.of();
+        }
+        final ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        list.forEach(o -> {
+            try {
+                builder.add(BeanUtils.getProperty(o, property));
+            } catch (final IllegalAccessException|InvocationTargetException |NoSuchMethodException e) {
+                LOG.warn("Cannot create property union of objects from class {} using property {}: {}",
+                        o.getClass().toString(),
+                        property,
+                        e.getMessage());
+            }
+        });
+
+        return builder.build();
+    }
+
+    /**
+     * Creates a map between the specified property in string form, and the object containing the property.
+     */
+    static <T> ImmutableMap<String, T> toPropertyMap(final ImmutableList<T> objects, final String property) {
+        if (isEmpty(objects)) {
+            return ImmutableMap.of();
+        }
+        final ImmutableMap.Builder<String, T> builder = ImmutableMap.builder();
+
+        objects.forEach(o -> {
+            try {
+                builder.put(BeanUtils.getProperty(o, property), o);
+            } catch (final IllegalAccessException|InvocationTargetException|NoSuchMethodException e) {
+                LOG.warn("Cannot create property map of object from class {} using property {}: {}",
```

Comment:
Make sure the exception is captured in the log message so that way the stack trace is included in the log message